cli/src/pcluster/validators/ebs_validators.py

```diff
@@ -199,18 +199,27 @@ class MultiAzEbsVolumeValidator(Validator):
     def _validate(self, head_node_az: str, ebs_volumes, queues):
         cross_az_queues = set()
         for volume in ebs_volumes:
-            if volume["az"] != head_node_az:
-                self._add_failure(
-                    "Your configuration includes an EBS volume '{0}' created in a different availability zone than "
-                    "the Head Node. The volume and instance must be in the same availability "
-                    "zone.".format(volume["name"]),
-                    FailureLevel.ERROR,
-                )
+            try:
+                # if the EBS volume is managed we set the AZ == to the HeadNode AZ otherwise we ask EC2 about
+                # the AZ where the existing volume is created
+                ebs_az = head_node_az if volume.is_managed else volume.availability_zone
+                if ebs_az != head_node_az:
+                    self._add_failure(
+                        "Your configuration includes an EBS volume '{0}' created in a different availability zone than "
+                        "the Head Node. The volume and instance must be in the same availability "
+                        "zone.".format(volume.name),
+                        FailureLevel.ERROR,
+                    )
 
-            for queue in queues:
-                queue_az_set = set(queue.networking.az_list)
-                if len(queue_az_set) > 1 or (set([volume["az"]]) != queue_az_set):
-                    cross_az_queues.add(queue.name)
+                for queue in queues:
+                    queue_az_set = set(queue.networking.az_list)
+                    if len(queue_az_set) > 1 or ({ebs_az} != queue_az_set):
+                        cross_az_queues.add(queue.name)
+            except AWSClientError as e:
+                if str(e).endswith("parameter volumes is invalid. Expected: 'vol-...'."):
+                    self._add_failure(f"Volume {volume.volume_id} does not exist.", FailureLevel.ERROR)
```

Comment:
surround `{volume.volume_id}` with single quote, to be consistent with other messaging